### PR TITLE
KAFKA-5374. AdminClient gets "server returned information about unkno…

### PR DESCRIPTION
**Pull request by cmccabe**
…wn correlation ID" when communicating with older brokers

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ static KafkaAdminClient createInternal(AdminClientConfig config) {
 
         try {
             metadata = new Metadata(config.getLong(AdminClientConfig.RETRY_BACKOFF_MS_CONFIG),
-                    config.getLong(AdminClientConfig.METADATA_MAX_AGE_CONFIG), false);
+                    config.getLong(AdminClientConfig.METADATA_MAX_AGE_CONFIG), true);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:285]: Can we please add a comment here stating that even though we allow auto creation, we never set `topics`, so we never trigger it.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. Just a minor comment.

**Comment by ijuma** [comment]
There are some checkstyle failures.

**Comment by ijuma** [comment]
@cmccabe, please close this PR. The change has been merged via #3220.

**Comment by cmccabe** [comment]
Closing as duplicate

**Pull request closed without merging**